### PR TITLE
Small refactoring and fix

### PR DESCRIPTION
**Pull request by dam1r89**
I'm still researching this. So far I've seen that `updateVisibleItems` is called multiple times on initial load which helps it to hide this issue.

When `updateVisibleItems` is called first time it has a bug. For first item it doesn't have available views in a pool, so it creates new unused view, for second item it gets view and adjust it to second item. Third item again doesn't have available view in the pool etc...
image

When `updateVisibleItems` is called second time everything is fine. Usually you don't see anything between calls because they are fast. In our case items render time is slow so you can clearly see this bug.
image

Basically this will create unused view and then assign item to it. This is probably fixing #368 

Also in the process I did small refactoring.

What do you think, am I understanding this correctly? Sorry for ugly images.

### CODE REVIEW

**Comment by gvn182** [comment]
Guys, is it fixed? im facing the same problem but scrolling to top, it is hard to reproduce but it happens very often.
